tests/unit/ui/components/TestShowMoreCard.js

```diff
@@ -1,105 +1,214 @@
   });
 
   it('truncates the contents if they are too long', () => {
-    const root = render({ children: 'Hello I am description' });
+    const { store } = dispatchClientMetadata();
+    const root = render({ store });
 
+    // We are simulating the truncate method call.
     root.instance().truncateToMaxHeight({ clientHeight: MAX_HEIGHT + 1 });
-    root.update();
+
+    applyUIStateChanges({ root, store });
 
     expect(root).not.toHaveClassName('ShowMoreCard--expanded');
   });
 
   it('renders className', () => {
     const className = 'test';
-    const root = render({
-      children: <p>Hi</p>,
-      className,
-    });
-
+    const root = render({ className });
     expect(root).toHaveClassName(className);
   });
 
   it('renders children', () => {
     const root = render({ children: 'Hello I am description' });
+    const contents = root.find('.ShowMoreCard-contents');
+    expect(contents).toHaveText('Hello I am description');
+  });
+
+  it("calls resetUIState if the children's html has changed", () => {
+    /* eslint-disable react/no-danger */
+    const root = render({
+      children: (
+        <div
+          dangerouslySetInnerHTML={{
+            __html: '<span>First component text.</span>',
+          }}
+        />
+      ),
+    });
+
+    const resetUIStateSpy = sinon.spy(root.instance(), 'resetUIState');
+
+    root.setProps({
+      children: (
+        <div
+          dangerouslySetInnerHTML={{
+            __html: '<span>This is different Text.</span>',
+          }}
+        />
+      ),
+    });
+    /* eslint-enable react/no-danger */
 
-    expect(root.childAt(0)).toHaveText('Hello I am description');
+    sinon.assert.called(resetUIStateSpy);
   });
 
-  it('executes truncateToMaxHeight when it recieves props', () => {
-    const root = mount(<ShowMoreCardBase i18n={fakeI18n()} />);
-    const component = root.instance();
+  it("calls resetUIState if the children's text has changed", () => {
+    const root = render({
+      children: 'Some text',
+    });
 
-    const contentNode = component.contents;
-    const truncateToMaxHeight = sinon.spy(component, 'truncateToMaxHeight');
-    root.setProps(); // simulate any kind of update to properties
+    const resetUIStateSpy = sinon.spy(root.instance(), 'resetUIState');
 
-    sinon.assert.calledWith(truncateToMaxHeight, contentNode);
+    root.setProps({
+      children: 'Some new text',
+    });
+
+    sinon.assert.called(resetUIStateSpy);
   });
 
-  it('expands if new child content is smaller', () => {
-    const root = mount(
-      <ShowMoreCardBase i18n={fakeI18n()}>
-        This would be very long content, but we cannot get `clientHeight` in the
-        tests, so this will be forced below (via setState()).
-      </ShowMoreCardBase>,
+  it("does not call resetUIState if the children's html is the same", () => {
+    /* eslint-disable react/no-danger */
+    const root = render({
+      children: (
```

Comment:
Because these values are the same, it might be better to set a `const` for `children` and just use that in both the `render` and the `setProps` calls.